Model/DutyCalculator.php

```diff
@@ -116,84 +116,333 @@ public function __construct(
         $this->_logger = $logger;
     }
 
+
+    /**
+     * Sets quote and checkout Session state based on whether duty and tax is applicable or not
+     * @param Magento\Quote\Api\Data\AddressInterface $address
+     * @param bool $apply  - whether duty is applicable or not
+     *
+     */
+    public function handleTaxApplicability($address, $apply)
+    {
+        $this->_logger->debug('In handleTaxApplicability method');
+        $this->_logger->debug($this->checkoutSession->getReachDuty());
+        $this->_logger->debug($address->getCountryId());
+        $this->_logger->debug($address->getRegionCode());
+        $this->_logger->debug($this->checkoutSession->getPrevCountry());
+        $this->_logger->debug($this->checkoutSession->getPrevRegion());
+        $this->_logger->debug($apply);
+        $quote = $this->checkoutSession->getQuote();
+
+        if ($apply || !$this->getIsOptional($address->getCountryId())) {
```

Comment:
these two logical cases follow a lot of the same logic with the same lines. Would it be clearer to set the ReachDuty to 0 if (!$apply || $this->getIsOptional($address->getCountryId)) and otherwise set the ReachDuty to $this->getIsOptional($address->getCountryId())  since that appears to be the only things that is different between the two cases?

But I might be reading this wrong, just wanted to check since it took me a bit to work out what was happening here.

Reply:
Thanks Lizzy. This https://github.com/withreach/magento-plugin/pull/44/commits/514a0df534ca810d3147db469740dedbcec9e79a commit hopefully handles it.